### PR TITLE
add ClusterColocationProfile CRD

### PR DESCRIPTION
**Pull request by eahydra**
Signed-off-by: Tao Li <joseph.t.lee@outlook.com>

### Ⅰ. Describe what this PR does
Add ClusterColocationProfile CRD

### CODE REVIEW

File: apis/config/v1alpha1/cluster_colocation_profile_types.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+	// ObjectSelector decides whether to mutate/validate Pods based on if the
+	// Pod has matching labels.
+	// Use the object selector only if the profile is opt-in, because end
+	// users may skip the admission webhook by setting the labels.
+	// Default to the empty LabelSelector, which matches everything.
+	// +optional
+	ObjectSelector *metav1.LabelSelector `json:"objectSelector,omitempty"`
+
+	// Labels describes the k/v pair that needs to  inject into Pod.Labels
```
Comment (hormes) [inline apis/config/v1alpha1/cluster_colocation_profile_types.go:43]: to inject，多了个空格

File: .gitignore [hunk 2]
```diff
@@ -26,3 +26,5 @@ testbin/*
 
 .vscode
 .DS_Store
+# Ignore this directory temporarily to bypass the check
+/vendor/
```
Comment (jasonliu747) [inline .gitignore:30]: we might need to keep vendor folder

File: apis/config/v1alpha1/cluster_colocation_profile_types.go (hunk 1 above)
Comment (eahydra) [reply apis/config/v1alpha1/cluster_colocation_profile_types.go]: done

File: .gitignore (hunk 2 above)
Comment (eahydra) [reply .gitignore:30]: Ignore this directory temporarily to bypass the check.
The vendor folder needs to be added via a separate PR.

**Review by hormes — APPROVED** [review]

**Pull request merged**